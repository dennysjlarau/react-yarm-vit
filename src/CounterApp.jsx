import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {
  const handleAdd = (event, value) => { 
    console.log(event);
    value = value + 1;
    console.log(value);
  }
  
  return (
    <>
      <h1>CounterApp</h1>
      <h2> { value } </h2>

      <button onClick={ (event) => handleAdd(event, value) }>
        +1
      </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number
}

CounterApp.defaultProps = {
  value: 0
}