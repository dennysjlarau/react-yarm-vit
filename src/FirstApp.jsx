// import { Fragment } from "react"

// export const FirstApp = () => {
//   return (
//     <Fragment>
//       <h2>Dennys Javier Lara Uquillas</h2>
//       <p>Este es un ejemplo de párrafo</p>
//     </Fragment>
//   )
// }

const newMessage = {
  message: 'Hola Mundo',
  title: 'Dennys Lara',
};

const getSuma = (num1, num2) => {
  return num1 + num2;
}

export const FirstApp = ({title='título por defecto'}) => {
  return (
    <>
      <code>{ JSON.stringify(newMessage) }</code>
      <p>Este es un ejemplo de párrafo</p>
      <h1>{getSuma(3,5)}</h1>
      <h2>{ title }</h2>
    </>
  )
}