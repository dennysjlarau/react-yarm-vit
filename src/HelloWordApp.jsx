//export function App(){
//    // documento.createElement...
//    return <h1>Hola Mundo</h1>;
//  }

export const HelloWordApp = () => {
  return (
    <h1>Hola Mundo App</h1>
  )
}
