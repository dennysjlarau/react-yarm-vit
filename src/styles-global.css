html, body {
    background-color: dimgray;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 45px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
    margin-right: 5px;
}