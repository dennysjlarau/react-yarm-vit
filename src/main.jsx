import React from 'react';
import ReactDOM from 'react-dom/client';
import {HelloWordApp} from './HelloWordApp'
import {FirstApp} from './FirstApp'

import './styles-global.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <FirstApp subTitle={123}/>
  </React.StrictMode>
);