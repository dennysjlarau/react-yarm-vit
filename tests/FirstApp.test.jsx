/**  * @jest-environment jsdom  */ 
import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas Unitarias: de FirstApp', () => {
    test('Prueba: del snapshot', () => {
        const title = 'título de prueba';
        const {container} = render(<FirstApp title = {title}/>);
        
        expect(container).toMatchSnapshot();

    });
    test('Prueba: debe mostrar el título en un h2', () => {
        const title = 'título de prueba';
        const {container, getByText} = render(<FirstApp title = {title}/>);
        
        expect(getByText(title)).toBeTruthy();

        const h2 = container.querySelector('h2');
        expect(h2.innerHTML).toContain(title);
    });
});