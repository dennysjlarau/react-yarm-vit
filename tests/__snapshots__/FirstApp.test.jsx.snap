// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Unitarias: de FirstApp Prueba: del snapshot 1`] = `
<div>
  <code>
    {"message":"Hola Mundo","title":"Dennys Lara"}
  </code>
  <p>
    Este es un ejemplo de párrafo
  </p>
  <h1>
    8
  </h1>
  <h2>
    título de prueba
  </h2>
  <h2>
    no hay subtitulo
  </h2>
</div>
`;
